src/compiler/bundle/app-data-plugin.ts

```diff
@@ -60,7 +62,22 @@ export const appDataPlugin = (
         appendEnv(config, s);
         return s.toString();
       }
-      return null;
+      if (id !== config.globalScript) {
+        return null;
+      }
+
+      const mod = compilerCtx.moduleMap.get(config.globalScript);
+      if (!mod.sourceMapFileText) return {code: mod.staticSourceFileText, map: null};
+
+      const sourceMap: d.SourceMap = JSON.parse(mod.sourceMapFileText);
+      const rollupSrcMap = {
+        mappings: sourceMap.mappings,
+        sourcesContent: sourceMap.sourcesContent,
+        sources: sourceMap.sources.map(src => basename(src)),
+        names: sourceMap.names,
+        version: sourceMap.version
+      };
+      return {code: mod.staticSourceFileText, map: rollupSrcMap};
```

Comment:
It would be awesome to get stronger typing on the return of this function here. I'm going to PR into your checkout, because I found one interesting thing that I have a question about that you may know better. 

Here's the signature of `Plugin#load` https://github.com/rollup/rollup/blob/master/docs/05-plugin-development.md#load I duplicated this into the definitions and called it RollupLoadHook. 

When I added that definition, almost everything passes except for the rollupSrcMap. the map property of the LoadHook is expecting the type of `d.SourceMap` (the `sourceMap` const), but you pass in the rollupSrcMap instead of the sourceMap. 

Just so I can understand how you're using this, how does this extra detail get consumed? Would we be able to put this inside LoadHook#meta and keep the sourceMap variable as the map value? 

See https://github.com/johnjenkins/stencil/pull/1/files

I'm just getting started too, so if I'm missing something, let me know!

Reply:
Ok, now that I've reviewed this and I've checked it out to review in a few live repos, I understand better what you're doing with the rollupSrcMap. I answered my own question - meta wouldn't work for what you're doing here. 

My two questions with this - and you have this same code in a couple places - why do you omit some of the properties from that `sourceMap` const? And what is the goal of updating the sources array with the basename? Is there some interop you get from doing that I may be missing? Sources seem to resolve correctly without that basename call.

Reply:
Ah, it's the file property of a source map that you're omitting here. Is that intentional? 

Then for the sources bit, could we rewrite this chunk as:

```javascript
      const sourceMap: d.SourceMap = JSON.parse(module.sourceMapFileText);
      sourceMap.sources = sourceMap.sources.map(src => basename(src));
      return { code: module.staticSourceFileText, map: sourceMap };
```

Added that in https://github.com/johnjenkins/stencil/pull/1/commits/9ef00edc04be7936a822f59a9bacb399d73074b5

Reply:
@splitinfinities I need to jog my memory on all this! 
But I think you're right - I was learning about source maps at the same time as doing this, and I have a vague feeling that I was kinda confused about `file` and how it would relate to the final module - especially as stencil changes some file names totally outside of rollup (e.g. https://github.com/ionic-team/stencil/blob/master/src/compiler/output-targets/dist-lazy/write-lazy-entry-module.ts#L17). As it didn't seem to make a difference to the end result opted (almost certainly incorrectly and without testing) to take it out. 
After some more prodding I see that for the aforementioned hashed re-named files - their source maps seem to lose the `file` attribute somewhere else along the road anyway - can't quickly see where.